components/builder-depot/src/config.rs

```diff
@@ -94,6 +99,35 @@ impl GatewayCfg for Config {
     }
 }
 
+#[derive(Debug, Clone, Deserialize)]
+pub enum S3Backend {
+    Aws,
+    Minio,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(default)]
+pub struct S3Cfg {
+    // These are for using S3 as the artifact storage
+    pub key_id: String,
+    pub secret_key: String,
+    pub bucket_name: String,
+    pub backend: S3Backend,
+    pub endpoint: String,
+}
+
+impl Default for S3Cfg {
+    fn default() -> Self {
+        S3Cfg {
+            key_id: String::from(""),
```

Comment:
Might be good to match all the defaults we put in the default.toml here as well..

Reply:
Yep I had this locally as well :)